File: algorithms/c#/InsertionSort.cs
namespace Algorithms
{
	public static class InsertionSort
	{
		public static void Sort(int[] array)
		{
			for(int i = 0; i < array.Length; i++)
			{
				int j = 0;
				int key = array[i];

				for(j = i - 1; j >= 0; j--)
				{
					if(array[j] > key) array[j + 1] = array[j];
					else break;
				}

				array[j + 1] = key;
			}
		}
	}
}

File: algorithms/go/insertion_sort.go
package algorithms

func InsertionSort(slice []int) {
	for i := 1; i < len(slice); i++ {
		j := i - 1
		key := slice[i]

		for j >= 0 && slice[j] > key {
			slice[j + 1] = slice[j]
			j--
		}

		slice[j + 1] = key
	}
}